database/migrations/default/1742408937781_crash_ems_link/up.sql

```diff
@@ -0,0 +1,391 @@
+                WHERE 
+                    c.is_deleted = false
+                    and not c.is_temp_record
+                    AND matching_ems.incident_received_datetime  >= (c.crash_timestamp - time_threshold)
+                    AND matching_ems.incident_received_datetime  <= (c.crash_timestamp + time_threshold)
+                    AND c.position IS NOT NULL
+                    AND ST_DWithin(matching_ems.geometry::geography, c.position::geography, meters_threshold)
+            ) INTO matched_crash_ids;
+            
+        -- Get the count from the array length (handling when array_length is null as 0)
+        SELECT COALESCE(array_length(matched_crash_ids, 1), 0) INTO match_count;
+        
+        IF match_count = 0 THEN
+            UPDATE ems__incidents 
+            SET crash_pk = NULL,
+                crash_match_status = 'unmatched',
+                matched_crash_pks = NULL
+            WHERE id = matching_ems.id;
+        ELSIF match_count = 1 THEN
+            -- this EMS record is only matched to one crash - we can assign the crash_pk
+            UPDATE ems__incidents 
+            SET crash_pk = NEW.id,
+                crash_match_status = 'matched_by_automation',
+                matched_crash_pks = NULL
+            WHERE id = matching_ems.id;
+        ELSE
+            -- multiple matching crashes found - cannot assign crash_pk
+            UPDATE ems__incidents 
+            SET crash_match_status = 'multiple_matches_by_automation',
+                crash_pk = NULL,
+                matched_crash_pks = matched_crash_ids
+            WHERE id = matching_ems.id;
+        END IF;
+    END LOOP;
+
+    IF TG_OP = 'UPDATE' THEN
+        UPDATE ems__incidents
+        SET crash_pk = NULL,
+            crash_match_status = 'unmatched',
+            matched_crash_pks = NULL
+        WHERE crash_pk = NEW.id
+          AND (
+              incident_received_datetime  < (NEW.crash_timestamp - time_threshold)
+              OR incident_received_datetime  > (NEW.crash_timestamp + time_threshold)
+              OR geometry IS NULL
+              OR NEW.position IS NULL
+              OR (NOT ST_DWithin(geometry::geography, NEW.position::geography, meters_threshold))
+          )
+          AND 
+            -- ignore manual qa matches: these should not be modified
+            crash_match_status != 'matched_by_manual_qa';
+    END IF;
+    RETURN NEW;
+END;
+$$ language plpgsql;
+
+
+--
+-- create insert trigger
+--
+create or replace trigger crash_insert_ems_match_trigger
+after insert on crashes
+for each row execute function update_crash_ems_match();
+
+--
+-- create update trigger
+--
+create or replace trigger crash_update_ems_match_trigger
+after update on crashes
+for each row when (
+    (new.latitude is distinct from old.latitude)
+    or (new.longitude is distinct from old.longitude)
+    or (new.crash_timestamp is distinct from old.crash_timestamp)
+) execute function update_crash_ems_match();
+
+
+--
```

Comment:
this trigger used to parse out the first to incident numbers and assign them to their own columns. thanks to our updated EMS query, it's really common for there to be more than two incident numbers, and their order is completely random. these columns no longer serve any useful purpose.